os/board/common/crashdump.c

```diff
@@ -53,11 +69,186 @@
 /****************************************************************************
  * Private Variables
  ****************************************************************************/
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH)
+static char ramdump_partname[MTD_PARTNAME_LEN + 1];
+static uint32_t ramdump_partoffset = 0;
+static uint32_t ramdump_partsize = 0;
+#endif
+
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+static char coredump_partname[MTD_PARTNAME_LEN + 1];
+static uint32_t coredump_partoffset = 0;
+static uint32_t coredump_partsize = 0;
+#endif
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
 
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH) || defined(CONFIG_BOARD_COREDUMP_FLASH)
+static int get_dump_partindex(char *part_name)
```

Comment:
It looks mtd functionality, not crash dump's. How about moving into mtd_partition.c with `int get_mtd_partitionindex(char *part_name)`?